lstm/bmi_lstm.py

```diff
@@ -1,1046 +1,562 @@
-        original: np.ndarray = self.get_value_ptr(var_name)
-        for i in range(indices.shape[0]):
-            value_index = indices[i]
-            dest[i] = original[value_index]
+    def get_var_location(self, name: str) -> str:
+        if name in StateFacade(self._outputs, self._dynamic_inputs):
+            return "node"
+        raise RuntimeError(f"unknown name: {name!s}")
+
+    def get_current_time(self) -> float:
+        return self._timestep * self._timestep_size_s
+
+    def get_start_time(self) -> float:
+        return 0
+
+    def get_end_time(self) -> float:
+        return np.finfo("d").max  # type: ignore
+
+    def get_time_units(self) -> str:
+        return "s"
+
+    def get_time_step(self) -> float:
+        return self._timestep_size_s
+
+    def get_value(self, name: str, dest: np.ndarray) -> np.ndarray:
+        dest[:] = self.get_value_ptr(name)
         return dest
- 
-    #   Note: remaining grid funcs do not apply for type 'scalar'
-    #   Yet all functions in the BMI must be implemented 
-    #   See https://bmi.readthedocs.io/en/latest/bmi.best_practices.html          
-    #------------------------------------------------------------ 
-    def get_grid_edge_count(self, grid):
-        raise NotImplementedError("get_grid_edge_count")
-
-    #------------------------------------------------------------ 
-    def get_grid_edge_nodes(self, grid, edge_nodes):
-        raise NotImplementedError("get_grid_edge_nodes")
-
-    #------------------------------------------------------------ 
-    def get_grid_face_count(self, grid):
-        raise NotImplementedError("get_grid_face_count")
-    
-    #------------------------------------------------------------ 
-    def get_grid_face_edges(self, grid, face_edges):
-        raise NotImplementedError("get_grid_face_edges")
-
-    #------------------------------------------------------------ 
-    def get_grid_face_nodes(self, grid, face_nodes):
-        raise NotImplementedError("get_grid_face_nodes")
-    
-    #------------------------------------------------------------ 
-    def get_grid_node_count(self, grid):
-        raise NotImplementedError("get_grid_node_count")
-
-    #------------------------------------------------------------ 
-    def get_grid_nodes_per_face(self, grid, nodes_per_face):
-        raise NotImplementedError("get_grid_nodes_per_face") 
-    
-    #------------------------------------------------------------ 
-    def get_grid_origin(self, grid_id, origin):
-        raise NotImplementedError("get_grid_origin") 
-
-    #------------------------------------------------------------ 
-    def get_grid_rank(self, grid_id):
- 
-         # 0 is the only id we have
-        if grid_id == 0: 
-            return 1
 
-    #------------------------------------------------------------ 
-    def get_grid_shape(self, grid_id, shape):
-        raise NotImplementedError("get_grid_shape") 
+    def get_value_ptr(self, name: str) -> np.ndarray:
```

Comment:
Probably need to include a little bit of the doc string, particularly on something like `get_value` and `get_value_ptr` to clearly document the difference in semantics between copying and referencing and WHAT is being copied/referenced (e.g. numpy array, scalar, scalar numpy array)

Reply:
Good call. Done